lib_standard_app/io.c

```diff
@@ -153,22 +153,27 @@ WEAK int io_recv_command()
     return ret;
 }
 
-WEAK int io_send_response_pointer(const uint8_t *ptr, size_t size, uint16_t sw)
-{
-    return io_send_response_buffer(&(const buffer_t){.ptr = ptr, .size = size, .offset = 0}, sw);
-}
-
-WEAK int io_send_response_buffer(const buffer_t *rdata, uint16_t sw)
+WEAK int io_send_response_buffers(const buffer_t *rdatalist, size_t count, uint16_t sw)
 {
     int ret = -1;
 
-    if (rdata != NULL) {
-        if (rdata->size - rdata->offset > IO_APDU_BUFFER_SIZE - 2 ||  //
-            !buffer_copy(rdata, G_io_apdu_buffer, sizeof(G_io_apdu_buffer))) {
-            return io_send_sw(SW_WRONG_RESPONSE_LENGTH);
+    if (rdatalist && count > 0) {
+        G_output_len = 0;
+        for (size_t i = 0; i < count; i++) {
+            const buffer_t *rdata = &rdatalist[i];
+
+            if (G_output_len + rdata->size - rdata->offset > IO_APDU_BUFFER_SIZE - 2
+                || !buffer_copy(rdata,
+                                G_io_apdu_buffer + G_output_len,
+                                sizeof(G_io_apdu_buffer) - G_output_len)) {
+                return io_send_sw(SW_WRONG_RESPONSE_LENGTH);
+            }
+            G_output_len += rdata->size - rdata->offset;
```

Comment:
Could add an overflow check here to avoid writing the status word at a bad place, or sending more than G_io_apdu_buffer